File: src/main/java/com/vizientinc/dungeonbase/models/Location.java
package com.vizientinc.dungeonbase.models;

import lombok.Data;

import java.util.List;

@Data
public class Location {
    String id;
    String name;
    String description;
    List<String> related;

    public Location(String name, String description, List<String> related){
        this.name = name;
        this.description = description;
        this.related = related;
    }
}

File: dotnet/Domains/Location.cs
using System.Collections.Generic;
using dungeonbase.Models;

namespace dungeonbase.Domains
{
    public class Location
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public List<LocationRecord> Related { get; set; }
    }
}